### ISSUE TITLE
connection drops with mysql causing error on first push

### ISSUE THREAD

**Issue by Brando47** [gotify/server#323]
**Can the issue be reproduced with the latest available release? (y/n)**
Yes I believe so, built from latest docker container

**Which one is the environment gotify server is running in?**
- [x] Docker
- [ ] Linux machine
- [ ] Windows machine
<details><summary>Docker startup command or config file here (please mask sensitive information)</summary><br>

```
docker run -d --name gotify \
  --restart unless-stopped \
  -p 2345:2345 \
  -v /app/gotify/data:/app/data \
  -e GOTIFY_SERVER_PORT=2345 \
  -e GOTIFY_DATABASE_DIALECT=mysql \
  -e GOTIFY_DATABASE_CONNECTION="gotifyUser:p4ssw0rd@tcp(10.2.2.245:3306)/gotify?charset=utf8mb4&collation=utf8mb4_unicode_ci&parseTime=True&loc=Local" \
  -e TZ=Australia/Adelaide \
  gotify/server
```
</details>

**Do you have an reverse proxy installed in front of gotify server? (Please select None if the problem can be reproduced without the presense of a reverse proxy)**
- [x] None
- [ ] Nginx
- [ ] Apache
- [ ] Caddy

**On which client do you experience problems? (Select as many as you can see)**
- [x] WebUI
- [x] gotify-cli
- [ ] Android Client 
- [x] 3rd-party API call (Please include your code)

```
curl -s -X POST "http://10.2.2.245:2345/message" -H 'X-Gotify-Key: mYaPi.KeY' -H 'Content-Type: application/json' -d '{ "title": "Test", "message": "Message", "priority": 6 }'
```

**What did you do?**
Attempts to communicate with the server after the connection to the mysql database has dropped will always result in an internal server error (500) the first time. This occurs with the API, gotify-cli (built from the latest commit) and the Web UI (just loading pages). I have been testing by posting a message but other endpoints that seem to require the database also reproduce the same error (as shown in the logs excerpt below). The second time a call is made works every time (even though there appears to be no open connection according to SHOW PROCESSLIST on the mysql cli), this works even if the usual "timeout" failure period has expired. Once the second call has been made, it will then work for as long as mysql holds the connection open, which is equal to the mysql environment variable 'wait_timeout'. The default for my MariaDB docker instance is 600 seconds, however raising it as far as 3600s is still producing the error. I figured this out when I found the following issue that seems very much related (although I am no expert/developer by any means): [https://github.com/go-sql-driver/mysql/issues/674](https://github.com/go-sql-driver/mysql/issues/674)

**What did you expect to see?**
Ideally the API should respond to the first attempt.

**What did you see instead? (Include screenshots, android logcat/request dumps if possible)**
In summary the API does not respond at all on the first attempt. On the second attempt, the connection becomes visible in mysql and it continues to work for wait_timeout seconds until the issue happens again. The error is shown as such in the logs:
```
[mysql] 2020/05/31 17:53:02 packets.go:36: unexpected EOF
[GIN] 2020/05/31 - 17:53:02 | 500 |     247.062µs |        10.2.2.1 | GET      "/application"
Error #01: an error occured while authenticating user

(/proj/database/client.go:24)
[2020-05-31 17:53:02]  invalid connection
```

Please let me know if I can provide any more information. Also wanted to say thanks for the work that's been done, it's very nice to be able to have an otherwise solid, professional looking self-hosted solution!

**Comment by jmattheis** [comment]
Hey @Brando47, thanks for the detailed bug report! Fix will be included in the next release.

**Comment by Brando47** [comment]
Speedy work, thank you so much!!
